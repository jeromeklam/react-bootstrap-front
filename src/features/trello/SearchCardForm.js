import React, { Component } from 'react';
import PropTypes from 'prop-types';

import EditableLabel from './EditableLabel';

import { Dropdown } from '../basic';

class SearchCardForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      ref: React.createRef(),
    };
    this.updateField = this.updateField.bind(this);
    this.handleAdd = this.handleAdd.bind(this);
    this.handleSelect = this.handleSelect.bind(this);
  }

  updateField = (field, value) => {
    this.setState({ [field]: value });
    this.props.onSearch(value);
  };

  handleAdd = () => {
    this.props.onAdd({ ...this.state, new: true, existing: false });
  };

  handleSelect = elem => {
    this.props.onAdd({ ...elem, new: false, existing: true });
  };

  render() {
    const { onCancel, t, data } = this.props;
    return (
      <div className="trello-search-card-form">
        <article className="trello-search-card-form-wrapper">
          <div ref={this.state.ref}>
            <EditableLabel
              placeholder={t('placeholder.num')}
              onKeyUp={e => this.updateField('num', e.target.value)}
              autoFocus
            />
          </div>
          {data && data.length > 0 && (
            <Dropdown
              className="border rounded border-secondary bg-white text-secondary"
              myRef={this.state.ref}
              maxHeight="250px"
            >
              {data.map(elem => (
                <a
                  href="/"
                  key={`trello-card-${elem.id}`}
                  className="dropdown-item"
                  onClick={e => {
                    if (e) {
                      e.preventDefault();
                      e.stopPropagation();
                    }
                    this.handleSelect(elem);
                  }}
                >
                  <span>{elem.label}</span>
                </a>
              ))}
            </Dropdown>
          )}
        </article>
        <button className="btn btn-primary" title={t('button.title.Add card')} onClick={this.handleAdd}>
          {t('button.Add card')}
        </button>
        <button className="btn btn-secondary" onClick={onCancel}>
          {t('button.Cancel')}
        </button>
      </div>
    );
  }
}

SearchCardForm.propTypes = {
  data: PropTypes.array,
  onSearch: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  onAdd: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

SearchCardForm.defaultProps = {
  data: [],
};

export default SearchCardForm;
