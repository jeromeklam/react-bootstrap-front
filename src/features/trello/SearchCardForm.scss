.trello-search-card-form {
  background-color: #eee;
}

.trello-search-card-form-wrapper {
  border-radius: 3px;
  background-color: #fff;
  position: relative;
  padding: 10px;
  cursor: pointer;
  max-width: 350px;
  margin-bottom: 7px;
  min-width: 330px;
  border: 1px solid;
  border-color: theme-color('secondary');
}