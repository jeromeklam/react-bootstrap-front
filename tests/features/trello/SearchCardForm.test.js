import React from 'react';
import renderer from 'react-test-renderer';
import SearchCardForm from '../../../src/features/trello/SearchCardForm';

const callback = jest.fn();

it('render with className and name', () => {
  // First render
  const component = renderer.create(
    <SearchCardForm className="btn bg-secondary" t={callback} />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
