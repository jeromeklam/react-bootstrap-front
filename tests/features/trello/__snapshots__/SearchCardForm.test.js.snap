// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with className and name 1`] = `
<div
  className="trello-search-card-form"
>
  <article
    className="trello-search-card-form-wrapper"
  >
    <div>
      <div
        className="comPlainTextContentEditable comPlainTextContentEditable--has-placeholder"
        contentEditable="true"
        onBlur={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onPaste={[Function]}
        placeholder=""
      />
    </div>
  </article>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  />
  <button
    className="btn btn-secondary"
  />
</div>
`;
